src/main/java/com/aws/greengrass/mqttbridge/MQTTBridge.java

```diff
@@ -87,35 +86,52 @@ protected MQTTBridge(Topics topics, TopicMapping topicMapping, MessageBridge mes
         this.pubSubClient = new PubSubClient(pubSubIPCAgent);
         this.ioTCoreClient = new IoTCoreClient(iotMqttClient);
         this.executorService = executorService;
-    }
 
-    @Override
-    public void install() {
-        mappingConfigTopics =
-                this.config.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY, MQTT_TOPIC_MAPPING);
-
-        mappingConfigTopics.subscribe((whatHappened, node) -> {
-            if (mappingConfigTopics.isEmpty()) {
-                logger.debug("Mapping empty");
-                topicMapping.updateMapping(Collections.emptyMap());
+        // handle configuration changes
+        Topics mappingConfigTopics = topics.lookupTopics(BridgeConfig.PATH_MQTT_TOPIC_MAPPING);
+        topics.lookupTopics(KernelConfigResolver.CONFIGURATION_CONFIG_KEY).subscribe((what, child) -> {
+            // initialization
+            if (child == null) {
+                onMqttTopicMappingChange(mappingConfigTopics);
+                return;
+            }
+
+            // handle mqtt topic mapping changes dynamically
+            if (child.childOf(BridgeConfig.KEY_MQTT_TOPIC_MAPPING)) {
+                onMqttTopicMappingChange(mappingConfigTopics);
                 return;
             }
 
-            try {
-                Map<String, TopicMapping.MappingEntry> mapping = OBJECT_MAPPER
-                        .convertValue(mappingConfigTopics.toPOJO(),
-                                new TypeReference<Map<String, TopicMapping.MappingEntry>>() {
-                                });
-                logger.atInfo().kv("mapping", mapping).log("Updating mapping");
-                topicMapping.updateMapping(mapping);
-            } catch (IllegalArgumentException e) {
-                logger.atError("Invalid topic mapping").kv("TopicMapping", mappingConfigTopics.toString()).log();
-                // Currently, Nucleus spills all exceptions in std err which junit consider failures
-                serviceErrored(String.format("Invalid topic mapping. %s", e.getMessage()));
+            // otherwise, reinstall to completely refresh this plugin
+            if (Arrays.stream(BridgeConfig.ALL_KEYS).anyMatch(child::childOf)) {
+                logger.atInfo().kv("config-key", child.getName())
+                        .log("configuration change detected, reinstalling bridge");
```

Comment:
don't log this, this will actually happen more than once; you'll be notified for every single key that changed, so maybe 4 times or so this will log unnecessarily. We'll see the component reinstall itself from the logs so I think this is superfluous.